code/ai/ai_profiles.cpp

```diff
@@ -770,6 +772,9 @@ void ai_profile_t::reset()
 	if (mod_supports_version(3, 6, 10)) {
 		flags.set(AI::Profile_Flags::Reset_last_hit_target_time_for_player_hits);
 	}
+	if (mod_supports_version(3, 6, 14)) {
+		flags.set(AI::Profile_Flags::Carry_shield_difficulty_scaling_bug);
+	}
```

Comment:
Shouldn't this not apply to mods targeting 23.0?

Reply:
This bug has existed since 3.6.14 and has been affecting any mods unintentionally balanced with it in mind, so unless it is applied retroactively, it could affect their behavior. This is *really* retroactive application, but it's as close as we can get.

Reply:
No, I perfectly understand why it should apply to mods targeting older versions; I just don't understand why this conditional is left open so that mods targeting 23.0 default to buggy behavior.

Reply:
Oh! I misunderstood, why does not *not* apply to 23.0. To be honest, I don't know why I hadn't even considered a slightly more complicated check, I'm glad you mentioned it because I was somewhat upset to 'introduce' an opt-out bug to new verisons.